src/main/utility/mod.rs

```diff
@@ -19,23 +19,72 @@ pub mod stream_len;
 pub mod syscall;
 pub mod time;
 
+use crate::{core::worker::Worker, host::host::HostId};
+
 /// A type that allows us to make a pointer Send + Sync since there is no way
 /// to add these traits to the pointer itself.
 #[derive(Clone, Copy, Debug)]
-pub struct SyncSendPointer<T>(pub *mut T);
+pub struct SyncSendPointer<T>(*mut T);
 
 unsafe impl<T> Send for SyncSendPointer<T> {}
 unsafe impl<T> Sync for SyncSendPointer<T> {}
 
 impl<T> SyncSendPointer<T> {
-    /// Get the pointer.
+    /// SAFETY: The object pointed to by `ptr` must actually be `Send` and
+    /// `Sync`, or else not subsequently used in contexts where it matters.
+    pub unsafe fn new(ptr: *mut T) -> Self {
+        Self(ptr)
+    }
+
     pub fn ptr(&self) -> *mut T {
         self.0
     }
+}
+
+/// A type that allows us to make a pointer Send + Sync since there is no way
+/// to add these traits to the pointer itself.
+#[derive(Clone, Copy, Debug)]
+pub struct HostTreePointer<T> {
```

Comment:
Maybe a comment about what makes this different from `SyncSendPointer`?

Reply:
Done